bindings/cpp/CMakeLists.txt

```diff
@@ -0,0 +1,66 @@
+cmake_minimum_required(VERSION 3.12)
+cmake_policy(VERSION 3.12)
+
+project(PortAudio LANGUAGES CXX)
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
+
+# Todo (multi-generator): Add support for multiple generators like:
+# - {Debug, Release} x {Static, Dynamic} x {MT, MD (Windows only)}
+set(portaudio-cpp-sources)
+add_subdirectory(source/portaudiocpp) # define source-files
+
+add_library(portaudiocpp ${portaudio-cpp-sources})
+add_library(PortAudio::portaudiocpp ALIAS portaudiocpp) # For subdirectory build
+
+# Remove, when PortAudio::portaudio-cpp has it's own CMakeLists.txt
+if(TARGET portaudio)
+    add_library(PortAudio::portaudio ALIAS portaudio) # Todo (responsibility): move to PortAudio's own CMakeLists.txt
+elseif(NOT TARGET PortAudio::portaudio)
+    find_package(portaudio CONFIG REQUIRED) # Import, if not already existing (e.g. in a sub-project)
+
+    if(TARGET portaudio)
+        add_library(PortAudio::portaudio ALIAS portaudio) # Todo (responsibility): move to PortAudio's own CMakeLists.txt
+    endif()
+endif()
+
+target_link_libraries(portaudiocpp PUBLIC PortAudio::portaudio)
+target_include_directories(portaudiocpp PUBLIC
+    $<INSTALL_INTERFACE:include>
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+)
+target_compile_features(portaudiocpp PUBLIC cxx_std_14) # There is no benefit of using c++11
+
+### Export ###
+
+include(GNUInstallDirs)
+
+install(TARGETS portaudiocpp
+    EXPORT PortAudioCppTargets
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+    INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+)
+
+# Todo(export) copy header files:
+file(GLOB_RECURSE portaudiocpp-header-files include/*.hxx) # define header-files
```

Comment:
https://cmake.org/cmake/help/latest/command/file.html#glob-recurse
> We do not recommend using GLOB to collect a list of source files from your source tree. If no CMakeLists.txt file changes when a source is added or removed then the generated build system cannot know when to ask CMake to regenerate. The CONFIGURE_DEPENDS flag may not work reliably on all generators, or if a new generator is added in the future that cannot support it, projects using it will be stuck. Even if CONFIGURE_DEPENDS works reliably, there is still a cost to perform the check on every rebuild.

Reply:
This is only relevant, when a file in the GLOB is removed, added, or moved. This is often not relevant for developers, because they often call cmake __before every__ build.

In this case it is irrelevant, because those files are collected only for installation.

Reply:
> This is often not relevant for developers

It's not relevant until it is relevant and someone gets confused. Please list the files individually; it is more robust.

Reply:
Done, but I disagree in this case, developers usually don't control, if a header is in the install set. Therefore, it would be more robust the other way round.